utils/agent_display_console.py

```
@@ -105,3 +106,59 @@ async def select_prompt_console(self):
         write_constants_to_file()
 
         return task
+
+    async def prompt_for_tool_call_approval(self, tool_name: str, tool_args: dict, tool_id: str) -> dict:
+        self.console.print(Panel(f"🧞 Assistant proposes to call tool: [bold cyan]{tool_name}[/bold cyan]", title="Tool Call Proposed", expand=False))
+
+        current_args_json = json.dumps(tool_args, indent=2)
```

Ruff (0.11.9)

Undefined name `json`

(F821)